### PR TITLE
Delphi Mock Overview and Coffee maker sample

### PR DESCRIPTION
**Pull request by bogdanpolak**

### CODE REVIEW

**Comment by bogdanpolak** [comment]
Please do not merge this branch yet. I added  requested this to discuss content. I want to add Coffee Maker sample (in progress now) before merge to master

File: README.md [hunk 1]
```diff
@@ -1,8 +1,36 @@
...
-# Example
+## Overview
+
+Delphi Mocks is a mocking framework that allows to replace class dependencies with a light implementation of it. This light objects are called test doubles and they are used in unit testing.
+
+Automated testing is mandatory this days and usually technical debt is measured with unit tests coverage. High test coverage allows to apply advanced refactorings and improve even very old code. However in object-oriented world, writing high quality unit test is challenging because of the composition. Object composition is the most popular way of building complex systems. It increases code reusability, but also makes automated testing much more difficult. Why? Because methods of such composed object can not be executed without its dependencies. Delphi Mocks framework enables dynamic substitution of this dependency and provides the opportunity to improve code test coverage.
+
+## Coding to the interfaces
+
+Known also as interface-based programming is a technique for writing interface-based classes and creating composed classes from interfaces - usually uses dependency injection to provide these dependent interfaces to encapsulated private fields of the class.
```
Comment (bogdanpolak) [inline README.md]: > usually uses dependency injection to provide these dependent interfaces to encapsulated private fields of the class.
Now it tools for me like a little to compressed, what maybe makes this sentence to difficult to understand. But fro the other side my idea was to keep overview as much compressed as it is possible. Any suggestions

Comment (vincentparrett) [reply README.md]: I would avoid going into too much detail here about DI etc. The main purpose of mocking is to do Isolation Testing - ie ensure that you are only testing a very small part of the application in isolation, everything else needed to stand up that part is mocked - take a look at this - https://www.typemock.com/general-unit-testing-page-page3/

**Comment by bogdanpolak** [comment]
I fixed overview message to be more isolation oriented.
And I'm going to finish my CoffeeMaker sample and put portion of it under the overview.

Sample preview is here: https://github.com/bogdanpolak/coffee-maker

I want to start with that code:

```pas
// aMachineTester: TStub<IMachineTester>;
// aCoffee: TCoffee;
aMachineTester := TStub<IMachineTester>.Create;
aMachineTester.Setup.WillReturn(True).When.IsReadyToBrewCoffee;
fCoffeeMachine := TCoffeeMachine.Create(
  {} TMock<IBrewingUnit>.Create,
  {} TMock<IGrinder>.Create,
  {} TMock<IUserPanel>.Create,
  {} aMachineTester);
aCoffee := fCoffeeMachine.BrewCoffee(csEspresso);
writeln(aCoffee.ToString);
```

**Comment by bogdanpolak** [comment]
Last three months was too difficult for me and I stoped all my OpenSource effort. Now thanks the general relaxation of restrictions I fill good energy to get back to GitHub contribution. I will reduce size of this PR and number of updates in main README. You can expect to see amendments soon. :)

**Comment by bogdanpolak** [comment]
*Note*: Add comment about compiler warnings: #106

**Pull request closed without merging**